### PR TITLE
Adding clock sync accuracy verification via metrics

### PR DESCRIPTION
**Pull request by edcdavid**
Uses Metrics to check the accuracy of test clock
MAX_OFFSET_IN_NS configures the offset threshold (default 300)
Proposal: I disabled interface test because they are too slow. We could re-implement this test using the existing l2 discovery data in a future PR

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ To run the conformance tests, first set the following environment variables:
 - **ENABLE_TEST_CASE**: This is an option to run the long running tests separated by comma. For example, to run reboot test, `ENABLE_TEST_CASE=reboot` shall be used.
 - **SKIP_INTERFACES**: passes a list of interfaces to be skipped in the form "eno1,ens2f1" `SKIP_INTERFACES="eno1,ens2f1"`
 - **KEEP_PTPCONFIG**: if set to true, the test script will not delete the ptpconfig it automatically created. If set to false, it will delete them. The ptpconfigs are deleted by default. For instance `KEEP_PTPCONFIG=true`
+- **MAX_OFFSET_IN_NS**: maximum offset in nanoseconds between a master and a slave clock when testing clock accuracy. Default is 300
```
Comment (aneeshkp) [inline test/README.md:11]: OFFSET ranges between MAX and MIN

File: test/utils/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	"github.com/openshift/ptp-operator/test/utils/client"
+	"github.com/openshift/ptp-operator/test/utils/pods"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	OpenshiftPtpInterfaceRole = "openshift_ptp_interface_role"
+	OpenshiftPtpOffsetNs      = "openshift_ptp_offset_ns"
+	OpenshiftPtpThreshold     = "openshift_ptp_threshold"
```
Comment (aneeshkp) [inline test/utils/metrics/metrics.go:22]: this is an range

File: test/utils/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+	metricsEndPoint           = "127.0.0.1:9091/metrics"
+	MaxOffsetDefaultNs        = 300
```
Comment (aneeshkp) [inline test/utils/metrics/metrics.go:24]: max/min

File: test/utils/metrics/metrics.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+	for index := range ptpPods.Items {
+
+		role, err := pods.PodRole(&ptpPods.Items[index], label)
+		if err != nil {
+			logrus.Errorf("cannot check pod role with err:%s", err)
+		}
+		if !role {
+			continue
+		}
+		return ptpPods.Items[index].Spec.NodeName, nil
```
Comment (aneeshkp) [inline test/utils/metrics/metrics.go:120]: in multi node cluster you are  returning only one Nodename?

File: test/utils/metrics/metrics.go (hunk 2 above)
Comment (edcdavid) [reply test/utils/metrics/metrics.go:22]: fixed

File: test/utils/metrics/metrics.go (hunk 3 above)
Comment (edcdavid) [reply test/utils/metrics/metrics.go]: fixed

File: test/README.md (hunk 1 above)
Comment (edcdavid) [reply test/README.md]: fixed

**Review by aneeshkp — APPROVED** [review]

**Pull request merged**